front/src/applications/editor/tools/rangeEdition/speedSection/SpeedSectionEditionLayers.tsx

```diff
@@ -95,16 +110,14 @@ export const SpeedSectionEditionLayers = () => {
     return featureCollection([...trackRangeFeatures, ...pslSignFeatures]);
   }, [entity, highlightedRoutes, trackSectionsCache]);
 
+  // Where the speed limit applies (entity track_ranges)
   const speedSectionsFeature: FeatureCollection = useMemo(() => {
     const flatEntity = flattenEntity(entity);
     // generate trackRangeFeatures
     const trackRanges = entity.properties?.track_ranges || [];
-    const trackRangeFeatures = trackRanges.flatMap((range, i) => {
-      const trackState = trackSectionsCache[range.track];
-      return trackState?.type === 'success'
-        ? getTrackRangeFeatures(trackState.track, range, i, flatEntity.properties)
-        : [];
-    }) as Feature<LineString | Point>[];
+    const trackRangeFeatures = trackRanges.flatMap(makeTrackRangeFeatures(flatEntity)) as Feature<
+      LineString | Point
+    >[];
     return featureCollection([...trackRangeFeatures]);
```

Comment:
No need to destructure here
```suggestion
    return featureCollection(trackRangeFeatures);
```

Reply:
done